specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2629 @@
+          "201": {
+            "description": "The request was successfully accepted; the operation will complete asynchronously. The provisioning state of the resource should indicate the current state of the resource.",
+            "schema": {
+              "$ref": "#/definitions/Task"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed. If the registry doesn't exist, 404 (Not found) is returned. If any of the input parameters is wrong, 400(Bad Request) is returned."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Tasks_Update": {
+            "$ref": "./examples/TasksUpdate.json"
+          },
+          "Tasks_Update_WithKeyVaultCustomCredentials": {
+            "$ref": "./examples/ManagedIdentity/TasksUpdate_WithKeyVaultCustomCredentials.json"
+          },
+          "Tasks_Update_WithMSICustomCredentials": {
+            "$ref": "./examples/ManagedIdentity/TasksUpdate_WithMSICustomCredentials.json"
+          },
+          "Tasks_Update_WithOpaqueCustomCredentials": {
+            "$ref": "./examples/TasksUpdate_WithOpaqueCustomCredentials.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/tasks/{taskName}/listDetails": {
+      "post": {
+        "tags": [
+          "Tasks"
+        ],
+        "description": "Returns a task with extended information that includes all secrets.",
+        "operationId": "Tasks_GetDetails",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TaskNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/Task"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed. If the registry doesn't exist, 404 (Not found) is returned."
+          }
+        },
+        "x-ms-examples": {
+          "Tasks_GetDetails": {
+            "$ref": "./examples/TasksGetDetails.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RunRequest": {
+      "description": "The request parameters for scheduling a run.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "The type of the run request.",
+          "type": "string",
+          "readOnly": true
```

Comment:
@shahzzzam - There was a bug in the model validator where it was throwing `INTERNAL_ERROR` instead of the actual error. That bug has been fixed. I just re-run the model validator CI and it is showing correct errors. Please take a look at the CI log over [here]( Tasks_Create_WithSystemAndUserIdentities).

There are basically two types of ERRORs:
1. READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST
2. OBJECT_ADDITIONAL_PROPERTIES

I am still looking into the 2nd type of error to ensure that it is not a false positive.
The 1st error `READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST` is correctly reported and you should be fixing it. 

- The property `"type"` is a discriminator and is not specified in the list of required properties of the model "RunRequest"
A property that is specified as a discriminator **MUST be a required property** of the model. Take a look at the `definition of discriminator` in Open API Specification 2.0 over [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-13) and `example model definition with discriminator property` over [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#models-with-polymorphism-support).
2.  The discriminator property `"type"` is also marked as `"readOnly": true`. Please read the definition of readOnly over [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-13). This model is then used in the request body for operation `Registries_ScheduleRun`. The example specifies the discriminator property "type" in the request body and thus the validator fails with `READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST`. If you do not specify the discriminator property then how is the sdk or the user supposed to serialize the model as a Parent `"RunRequest"` or `oneOf` its children `DockerBuildRequest`, `TaskRunRequest`, `FileTaskRunRequest`, `EncodedTaskRunRequest`. This can only be identified by the value provided in the discriminator property `"type"`.

Please make sure that you remove `readOnly: true` for all the properties marked as `discriminator` and add those properties to the list of required properties in the corresponding model definitions.

Reply:
@amarzavery Thanks for the detailed explanation on this. Quick question tho, the last api version was 2019-04-01. And I just copied over those examples from there. I am not sure how the previous api version did not detect such problems? The changes to the new API version aren't even related to all this.

Reply:
We improved our validation effort off lately especially the rule to ensure readonly properties are not allowed in the request. That opened the pandora's box of undetected issues.